awswrangler/s3/_select.py

```diff
@@ -0,0 +1,215 @@
+"""Amazon S3 Select Module (PRIVATE)."""
+
+import concurrent.futures
+import itertools
+import json
+import logging
+import pprint
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import boto3
+import pandas as pd
+
+from awswrangler import _utils, exceptions
+from awswrangler.s3._describe import size_objects
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+_RANGE_CHUNK_SIZE: int = int(1024 * 1024)
+
+
+def _select_object_content(
+    args: Dict[str, Any],
+    client_s3: boto3.Session,
+    scan_range: Optional[Tuple[int, int]] = None,
+) -> pd.DataFrame:
+    if scan_range:
+        args.update({"ScanRange": {"Start": scan_range[0], "End": scan_range[1]}})
```

Comment:
Looks kind of dangerous... (not thread-safe). Can we avoid mutating the args dict?

Reply:
Fair enough, I will split the calls